src/librustc/diagnostics.rs

```diff
@@ -176,6 +176,38 @@ for the entire lifetime of a program. Creating a boxed value allocates memory on
 the heap at runtime, and therefore cannot be done at compile time.
 "##,
 
+E0011: r##"
+Using user-defined operator on const/static variable is restricted to what can be
+evaluated at compile-time. Using an user-defined operator could call a user-defined
+function, which is not allowed.
+
+Bad example:
+
+```
+use std::ops::Index;
+
+struct Foo { a: u8 }
+ 
+impl ::std::ops::Index<u8> for Foo {
+    type Output = u8;
+ 
+    fn index<'a>(&'a self, idx: u8) -> &'a u8 { &self.a }
+}
+
+const a: Foo = Foo { a: 0u8};
+const b: u8 = a[0]; // Index trait is defined by the user, bad !
```

Comment:
All other Rust documentation I've read so far does not add spaces before exclamation marks.